src/state/tasks/tasksActionCreator.js

```diff
@@ -2,14 +2,44 @@ import * as types from './tasksActionTypes';
 import codeClanApi from '../../api/apiUtils';
 
 export const getAllTasksAction = trackId => {
-  return dispatch => {
+  return async dispatch => {
     dispatch({ type: types.TASKS_START });
     return codeClanApi
       .get('/tasks')
       .then(res => {
         const tasks = res.data.items.filter(task => task.track === trackId);
-        const tasksObj = { items: tasks, totalCount: tasks.length };
-        dispatch({ type: types.TASKS_SUCCESS, payload: tasksObj });
+        const submittedTasksUnfiltered = async () => {
+          return Promise.all(tasks.map(task => checkTaskSubmission(task)));
+        };
+        // const submittedTasks = submittedTasksUnfiltered().then(
+        //   res => res !== undefined
+        // );
+        const submittedTasks = submittedTasksUnfiltered()
```

Comment:
Codacy found an issue: ['submittedTasks' is assigned a value but never used.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=6160785)